Shared/Samples/Navigate route/NavigateRouteView.swift

```diff
@@ -0,0 +1,354 @@
+                
+                // Enables the navigate button.
+                isNavigateDisabled = false
+            } catch {
+                self.error = error
+            }
+        }
+        
+        /// Sets the route tracker and location display's data source with a solved route. Updates
+        /// the list of directions, the route ahead graphic, and the map's viewpoint.
+        func setNavigation() {
+            guard let routeResult else { return }
+            
+            // Creates a route tracker from the route results.
+            routeTracker = RouteTracker(routeResult: routeResult, routeIndex: 0, skipsCoincidentStops: true)
+            guard let routeTracker else { return }
+            
+            routeTracker.voiceGuidanceUnitSystem = Locale.current.usesMetricSystem ? .metric : .imperial
+            
+            // Gets the route and its geometry from the route result.
+            guard let firstRoute = routeResult.routes.first,
+                  let routeGeometry = firstRoute.geometry else {
+                return
+            }
+            
+            // Updates the directions.
+            directions = firstRoute.directionManeuvers
+            
+            // Creates the mock data source from the route's geometry.
+            let densifiedRoute = GeometryEngine.geodeticDensify(
+                routeGeometry,
+                maxSegmentLength: 50,
+                lengthUnit: .meters,
+                curveType: .geodesic
+            ) as! Polyline
+            let mockDataSource = SimulatedLocationDataSource(polyline: densifiedRoute)
+            
+            // Creates a route tracker location data source.
+            let routeTrackerLocationDataSource = RouteTrackerLocationDataSource(
+                routeTracker: routeTracker,
+                locationDataSource: mockDataSource
+            )
+            
+            // Sets the location display's data source.
+            locationDisplay.dataSource = routeTrackerLocationDataSource
+            
+            // Updates the graphics and viewpoint.
+            routeRemainingGraphic.geometry = routeGeometry
+            viewpoint = Viewpoint(boundingGeometry: routeGeometry)
+        }
+        
+        /// Start monitoring multiple asynchronous streams of information.
+        private func startTracking() async {
+            await withTaskGroup(of: Void.self) { group in
+                group.addTask { await self.trackAutoPanMode() }
+                group.addTask { await self.trackStatus() }
+                group.addTask { await self.trackVoiceGuidance() }
+            }
+        }
+        
+        /// Monitor the asynchronous stream of tracking statuses.
+        ///
+        /// When new statuses are delivered, update the route's traversed and remaining graphics.
+        private func trackStatus() async {
+            guard let routeTracker else { return }
+            for try await status in routeTracker.$trackingStatus {
```

Comment:
I attempted to remove the `try` keyword and the code still works. 🤔  Out of curiosity, can you find out why/when the tracking status can be `nil`?

In iOS API, the delegate method `func routeTracker(_ routeTracker: AGSRouteTracker, didUpdate trackingStatus: AGSTrackingStatus)` doesn't give a `nil` status.

Reply:
Hmm interesting. I see Kotlin generates the same code (navigation/RouteTracker.kt#L124), while in cocoa it only passes non optional status. If the status is `nil` only for the first empty location, then I think it's worth logging an issue to improve this API.

Reply:
After discussion, it is an expected behavior.